crates/sui-core/src/authority_active.rs

```diff
@@ -96,12 +97,11 @@ impl AuthorityHealth {
     }
 }
 
-#[derive(Clone)]
 pub struct ActiveAuthority<A> {
     // The local authority state
     pub state: Arc<AuthorityState>,
     // The network interfaces to other authorities
-    pub net: Arc<AuthorityAggregator<A>>,
+    pub net: ArcSwap<AuthorityAggregator<A>>,
```

Comment:
What we could do is instead of passing around `AuthorityActive` in all the active functions, we pass around `net`. Then we don't have to restart.
Another general question I have is this: why isn't the active processes unsafe with potential committee changes? Don't they need to deal with potential byzantine validator anyway?

Reply:
Passive processes should be able to handle a swap but active once are going to need some sort of notification in order to do internal book keeping.

Reply:
So really I think we're going to need to have a reconfig notification go out on a [broadcast channel](https://docs.rs/tokio/latest/tokio/sync/broadcast/index.html) as well to these arc swaps

Reply:
Not to mention that we cannot rely on ArcSwap in the future if we end up having multiple processes

Reply:
We have two active processes today: checkpoint and gossip.
checkpoint is not a concern because we do reconfiguration work from there, and we can control how to proceed after committee changes.
Gossip runs in an infinite loop, and one simple trick we could do is to just get the latest network at the beginning of the loop, and use that during the loop, so that nothing would change while it's actively running. However I do wonder why gossip would even care about committee changes in the middle of it. I would expect gossip to be able to handle it. Would love to know more about this.

So with above, at least today, I don't see a need for restart. Though it certainly can get complex if we plan to introduce a lot more active processes in the future

Reply:
The infra for notifying, and restarting the active processes is now here:
https://github.com/MystenLabs/sui/pull/2388

Reply:
we simply have to store the `LifecycleSignalSender` struct, and signal a `Restart` or `Exit`, and the services will restart (and this is when they should re-load the config etc) or exit.

Reply:
> Another general question I have is this: why are the active processes unsafe with potential committee changes? Don't they need to deal with potential byzantine validator anyway?

yeah, but they make the assumption that they know the correct committee and that 2/3 of it is correct. So if the committee changes half way through some task, and half the state is according to the old one, and the other half the new one, unknown things may happen.

Reply:
Reminder for myself: we probably should also add some safety guards around all validator requests on the epoch number: that is, regardless how we do this, we should be able to handle the case where you are in epoch X, talking to a committee member who thinks it's epoch X + 1.

Reply:
> I assume the gossip process doesn't have that assumption?

It kind of does: for example, it computes how long to wait for responses before trying to reconnect on the basis of the size and stake distribution of the committee to not spam everyone with connection requests before the network is up.